include/ck/utility/workgroup_synchronization.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+#include "ck/host_utility/hip_check_error.hpp"
+
+namespace ck {
+
+// Initialization flag of Barrier object, can be any value except for zero
+static constexpr int BarrierInitFlag = 0x7856;
+
+// only the first thread-block in the synchronizaton group is supposed to call this function
```

Comment:
How about if(block_local_id == 0) inside this function?

Reply:
That is not good,  because `block_local_id` is usually got in the calling context, not need to pass it into the primitive